MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/PolygonSelectToggleActivity.kt

```diff
@@ -0,0 +1,290 @@
+        setFeatureSelectState(feature, true)
+        refreshSource()
+    }
+
+    /**
+     * AsyncTask to load data from a GeoJSON file in the assets folder. Rather than loading from
+     * a locally-stored file, you could also use the Mapbox Dataset API or Tileset API to retrieve
+     * GeoJSON data.
+     */
+    private class LoadGeoJsonDataTask internal
+    constructor(activity: PolygonSelectToggleActivity) :
+            AsyncTask<Void, Void, FeatureCollection>() {
+
+        private val PROPERTY_SELECTED = "selected"
+        private val PROPERTY_FILL_COLOR = "fill_color"
+        private var randomColorListForPolygons = arrayOf("#20B2AA", "#A52A2A", "#FFB6C1", "#D2B48C",
+                "#FAFAD2", "#0000CD", "#FFFFF0", "#C71585", "#9400D3", "#F5F5DC", "#FFFACD", "#FFF5EE",
+                "#191970", "#FF8C00", "#000000", "#66CDAA", "#9ACD32", "#FFA07A", "#4B0082", "#E6E6FA"
+        )
+
+        private val activityRef: WeakReference<PolygonSelectToggleActivity>
+
+        init {
+            this.activityRef = WeakReference(activity)
+        }
+
+        override fun doInBackground(vararg params: Void): FeatureCollection? {
+            val activity = activityRef.get() ?: return null
+
+            val geoJson = loadGeoJsonFromAsset(activity,
+                    "new-orleans-neighborhoods.geojson")
+            return FeatureCollection.fromJson(geoJson)
+        }
+
+        override fun onPostExecute(featureCollection: FeatureCollection?) {
+            super.onPostExecute(featureCollection)
+            val activity = activityRef.get()
+            if (featureCollection == null || activity == null) {
+                return
+            }
+
+            for (singleFeature in featureCollection.features()!!) {
+                singleFeature.addBooleanProperty(PROPERTY_SELECTED, false)
+
+                // Add a String property to each Feature/neighborhood/polygon. The String value
+                // is the hex color value to be used in the `neighborhoodPolygonColoredFillLayer`
+                // fill layer.
+                singleFeature.addStringProperty(PROPERTY_FILL_COLOR,
+                        randomColorListForPolygons.get(Random().nextInt(
+                                randomColorListForPolygons.size)))
+            }
+            activity.setUpData(featureCollection)
+        }
+
+        companion object {
+            internal fun loadGeoJsonFromAsset(context: Context, filename: String): String {
+                try {
+                    // Load GeoJSON file
+                    val `is` = context.assets.open(filename)
+                    val size = `is`.available()
+                    val buffer = ByteArray(size)
+                    `is`.read(buffer)
+                    `is`.close()
+                    return String(buffer)
+                } catch (exception: Exception) {
+                    throw RuntimeException(exception)
+                }
```

Comment:
Reading files from assets in kotlin can be made simpler:
For example:
``` 
val file_name = "qjsonfile.json"
val json_string = application.assets.open(file_name).bufferedReader().use{
            it.readText()
        }
```

Reply:
Cool. Adjusted to:

```kotlin
companion object {
            internal fun loadGeoJsonFromAsset(context: Context, filename: String): String {
                try {
                    // Load GeoJSON file
                    context.assets.open(filename).bufferedReader().use{
                        return it.readText()
                    }
                } catch (exception: Exception) {
                    throw RuntimeException(exception)
                }
            }
        }
```